src/app/treasure-hunt/treasure-hunt-report/treasure-hunt-ppt.service.ts

```diff
@@ -0,0 +1,836 @@
+      let slideText: { text: pptxgen.TextProps[], options: pptxgen.TextPropsOptions } = this.getOpportunitySlideText(opp.opportunitySheet);
+      newSlide.addText(slideText.text, slideText.options);
+      newSlide.addText('Placeholder for picture', { x: 8.45, y: 1.2, w: 4.43, h: 2.81, align: 'center', fill: { color: 'BDEEFF' }, color: 'BFBFBF', fontSize: 18, fontFace: 'Arial (Body)', valign: 'middle', isTextBox: true, autoFit: true });
+      let rows = [];
+      rows.push([
+        { text: "Utility", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Energy Savings", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Unit", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Cost Saving ($)", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Material Cost ($)", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Labor Cost ($)", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Other Cost ($)", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Total Cost ($)", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } },
+        { text: "Payback (Years)", options: { color: "FFFFFF", bold: true, fill: { color: '1D428A' } } }
+      ]);
+      let x: OpportunitySummary = treasureHuntResults.opportunitySummaries[counter];
+      let utilityUnit: string;
+      if (x.mixedIndividualResults) {
+        x.mixedIndividualResults.forEach(x => {
+          utilityUnit = this.getUtilityUnit(x.utilityType, settings);
+          rows.push([x.utilityType, this.roundValToString(x.totalEnergySavings), utilityUnit, this.roundValToString(x.costSavings), this.roundValToString(x.opportunityCost.material), this.roundValToString(x.opportunityCost.labor), this.getOtherCost(x.opportunityCost), this.roundValToString(x.totalCost), this.roundValToString(x.payback)]);
+        });
+      } else {
+        utilityUnit = this.getUtilityUnit(x.utilityType, settings);
+        rows.push([x.utilityType, this.roundValToString(x.totalEnergySavings), utilityUnit, this.roundValToString(x.costSavings), this.roundValToString(x.opportunityCost.material), this.roundValToString(x.opportunityCost.labor), this.getOtherCost(x.opportunityCost), this.roundValToString(x.totalCost), this.roundValToString(x.payback)]);
+      }
+
+      newSlide.addTable(rows, { x: 1.14, y: 5.2, w: 11.05, colW: [1.5, 1.5, 0.8, 1.25, 1.25, 1.25, 1.25, 1.25, 1], color: "1D428A", fontSize: 12, fontFace: 'Arial (Body)', border: { type: "solid", color: '1D428A' }, fill: { color: 'BDEEFF' }, align: 'left', valign: 'middle' });
+      counter++;
+    });
+
+    return pptx;
+  }
+
+  roundValToString(num: number): string {
+    return Number(num.toFixed(2)).toLocaleString('en-US');
```

Comment:
toString() should work for you here. Read the doc on toLocaleString

Reply:
The toLocaleString('en-US') automatically adds comma formatting